src/networking/__tests__/VoiceWebSocket.test.ts

```diff
@@ -0,0 +1,121 @@
+import { VoiceOPCodes } from 'discord-api-types/v8/gateway';
+import EventEmitter, { once } from 'events';
+import WS from 'jest-websocket-mock';
+import { VoiceWebSocket } from '../VoiceWebSocket';
+
+beforeEach(() => {
+	WS.clean();
+});
+
+function onceIgnoreError<T extends EventEmitter>(target: T, event: string) {
+	return new Promise((resolve) => {
+		target.on(event, resolve);
+	});
+}
+
+function onceOrThrow<T extends EventEmitter>(target: T, event: string, after: number) {
+	return new Promise((resolve, reject) => {
+		target.on(event, resolve);
+		setTimeout(() => reject(new Error('Time up')), after);
+	});
+}
```

Comment:
Question about this, didn't we have an util to handle timed errors which cleans itself up (due to the lack of `timers/promises` in v14, which by the way, [there seems to have plans for a v14 backport](https://twitter.com/jasnell/status/1372163274699984906))

Reply:
We _do_, but then we have to rely on the implementation of the util being correct in the VoiceWebSocket test. Personally, I am not too bothered about the clean up in this instance since it's only running in a test suite and will be cleaned up anyway.